### PR TITLE
Add Jupyter Notebook exercises

### PR DESCRIPTION
**Pull request by jawnsy**

### CODE REVIEW

File: 02-hey-jupyter/README.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
+# Hey Jupyter
+
+## Introduction
+
+Jupyter Notebooks provide an interactive, web-based graphical interface for
+processing and analyzing data, similar to proprietary software products like
+MATLAB, Mathematica, and Maple. Jupyter is designed with with an architecture
+that supports multiple programming languages (the project name comes from
+[Julia](https://en.wikipedia.org/wiki/Julia_(programming_language),
```
Comment (Tom-Nudell) [inline 02-hey-jupyter/README.md]: missing closed parenthesis
[Julia](https://en.wikipedia.org/wiki/Julia_(programming_language))

File: 02-hey-jupyter/README.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+[Python](https://en.wikipedia.org/wiki/Python_(programming_language), and
```
Comment (Tom-Nudell) [inline 02-hey-jupyter/README.md]: missing closed parenthesis
`[Python](https://en.wikipedia.org/wiki/Python_(programming_language))`

File: 02-hey-jupyter/README.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+> a predecessor to Jupyter.
+
+### Count Records
+
+Let's determine the number of substations in this dataset. Filter the
+`terminals` data frame for entries with a `typ` of "substation":
+
+```
+substations = terminals[terminals["typ"] == "substation"]
+substations
```
Comment (Tom-Nudell) [inline 02-hey-jupyter/README.md]: I recommend getting in the habit of printing the head of data frames. Especially data frames of unknown size
`substations.head()`

File: 02-hey-jupyter/README.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+1. What are the two most common resistance/km values?
+
+### Derived Values
+
+We can derive new columns based on values of existing columns. For example,
+to determine total resistance of a given line segment, we can use:
+
+```
+branches["r_total"] = branches["r_ohmkm"] * (branches["length_m"] / 1000)
+branches
```
Comment (Tom-Nudell) [inline 02-hey-jupyter/README.md]: I recommend getting in the habit of printing the head of data frames. Especially data frames of unknown size
`branches.head()`

File: 02-hey-jupyter/README.md (hunk 3 above)
Comment (Tom-Nudell) [inline 02-hey-jupyter/README.md]: I recommend getting in the habit of printing the head of data frames. Especially data frames of unknown size
`substations.head()`

File: 02-hey-jupyter/README.md (hunk 2 above)
Comment (Tom-Nudell) [inline 02-hey-jupyter/README.md]: missing closed parenthesis
`[Python](https://en.wikipedia.org/wiki/Python_(programming_language))`

File: 02-hey-jupyter/README.md (hunk 1 above)
Comment (Tom-Nudell) [inline 02-hey-jupyter/README.md]: missing closed parenthesis
`[Julia](https://en.wikipedia.org/wiki/Julia_(programming_language))`

File: 02-hey-jupyter/README.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+```
+
+1. Which line has the largest total resistance?
+
+1. What are the lines with the 10 largest X/R ratios? 10 smallest?
+
+1. Which line has the highest total impedance? (magnitude only)
+   
+   _Hint_: recall that `Z = R + jX` and `|Z| = sqrt(R^2 + X^2)`
```
Comment (Tom-Nudell) [inline 02-hey-jupyter/README.md]: This hint probably isn't needed and might seem pedantic to a seasoned TPE

File: 02-hey-jupyter/README.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+
+### Create a Pull Request
+
+Save your Jupyter notebook with these exercises and create a Pull Request
```
Comment (Tom-Nudell) [inline 02-hey-jupyter/README.md]: Should we create a starter section for the 'Questions' in the base notebook?

File: 02-hey-jupyter/scigrid.ipynb [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "terminals = pd.read_csv('vertices_de_power_160718.csvdata', quotechar=\"'\")\n",
+    "terminals"
```
Comment (Tom-Nudell) [inline 02-hey-jupyter/scigrid.ipynb]: `terminals.head()`

File: 02-hey-jupyter/scigrid.ipynb [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "branches = pd.read_csv('links_de_power_160718.csvdata', quotechar=\"'\")\n",
+    "branches"
```
Comment (Tom-Nudell) [inline 02-hey-jupyter/scigrid.ipynb]: `branches.head()`

**Review by Tom-Nudell — CHANGES_REQUESTED** [review]

File: 02-hey-jupyter/README.md (hunk 1 above)
Comment (jawnsy) [reply 02-hey-jupyter/README.md]: Fixed, thanks!

File: 02-hey-jupyter/README.md (hunk 2 above)
Comment (jawnsy) [reply 02-hey-jupyter/README.md]: Fixed

File: 02-hey-jupyter/README.md (hunk 3 above)
Comment (jawnsy) [reply 02-hey-jupyter/README.md]: Great point. Fixed.

File: 02-hey-jupyter/README.md (hunk 4 above)
Comment (jawnsy) [reply 02-hey-jupyter/README.md]: Done

File: 02-hey-jupyter/README.md (hunk 5 above)
Comment (jawnsy) [reply 02-hey-jupyter/README.md]: Good point. Removed.

File: 02-hey-jupyter/scigrid.ipynb (hunk 7 above)
Comment (jawnsy) [reply 02-hey-jupyter/scigrid.ipynb]: Fixed

File: 02-hey-jupyter/scigrid.ipynb (hunk 8 above)
Comment (jawnsy) [reply 02-hey-jupyter/scigrid.ipynb]: Fixed

**Review by Tom-Nudell — APPROVED** [review]

**Pull request merged**